src/TheThingsNetwork.h

```diff
@@ -44,7 +44,7 @@ class TheThingsNetwork
     void init(Stream& modemStream, Stream& debugStream);
     void reset(bool adr = true, int sf = DEFAULT_SF, int fsb = DEFAULT_FSB);
     bool personalize(const byte devAddr[4], const byte nwkSKey[16], const byte appSKey[16]);
-    bool join(const byte appEui[8], const byte appKey[16]);
+    bool join(const byte appEui[8], const byte appKey[16], bool reset = true, long int nbr_delay = 20000, int max_attempts = 1000);
```

Comment:
- I would bring the delay down to `10000`.
- I would default `max_attempts` to `0` meaning it will try forever, which is what our examples currently do and the node is useless without join.